RHOSP-18.0.3/cinder/powerflex/README.md

```diff
@@ -0,0 +1,295 @@
+
+```bash
+$ touch /opt/emc/scaleio/openstack/connector.conf
+```
+Edit the `/opt/emc/scaleio/openstack/connector.conf` and populate it with passwords.
+
+Example:
+
+```
+[powerflex]
+san_password = powerflex_password
+
+[powerflex-new]
+san_password = powerflex_password
+```
+
+Before the dataplane is deployed, in the `dataplane-nodeset.yaml` enter the following
+```
+edpm_network_config_template: |
+...
+edpm_nova_extra_bind_mounts:
+  - src: /opt/emc/scaleio/openstack/connector.conf
+    dest: /opt/emc/scaleio/openstack/connector.conf
+    options: "ro"
+```
+
+Create PowerFlex connector secret 
+```yaml
+vi powerflex-connector-secret.yaml
+
+apiVersion: v1
+kind: Secret
+metadata:
+  labels:
+    component: cinder-volume
+    service: cinder
+  name: powerflex-connector-conf-files
+stringData:
+  connector.conf: |
+    [powerflex]
+    san_password=PF_Manager_password
+type: Opaque
+```
+
+Edit the existing OpenStackControlPlane CR and add an extraMount specification as follows
+```
+...  
+spec:
+  extraMounts:
+  - extraVol:
+    - extraVolType: powerflex
+      mounts:
+      - mountPath: /opt/emc/scaleio/openstack
+        name: powerflex-connector
+        readOnly: true
+      propagation:
+      - CinderVolume
+      volumes:
+      - name: powerflex-connector
+        secret:
+          secretName: powerflex-connector-conf-files
+    name: v1
+    region: r1
+ ...
+```
+
+### Test the configured Backend
+Finally, create a PowerFlex volume type and test if you can successfully create and attach volumes of that type.
+
+Run the following command to check whether the cinder-volume service is up. 
+```
+[admin@rhadmin ~]$ oc rsh openstackclient
+sh-5.1$ openstack volume service list
++------------------+-------------------------------------+------+---------+-------+----------------------------+
+| Binary           | Host                                | Zone | Status  | State | Updated At                 |
++------------------+-------------------------------------+------+---------+-------+----------------------------+
+| cinder-scheduler | cinder-scheduler-0                  | nova | enabled | up    | 2024-12-16T09:24:50.000000 |
+| cinder-volume    | cinder-volume-powerflex-0@powerflex | nova | enabled | up    | 2024-12-16T09:24:58.000000 |
+| cinder-backup    | cinder-backup-0                     | nova | enabled | up    | 2024-12-16T09:24:49.000000 |
++------------------+-------------------------------------+------+---------+-------+----------------------------+
+```
+Check if a volume type 'powerflex' is created.
+```
+sh-5.1$ openstack volume type list
++--------------------------------------+--------------------------------+-----------+
+| ID                                   | Name                           | Is Public |
++--------------------------------------+--------------------------------+-----------+
+| e8a9dcd6-e063-43ea-8132-e6e5e801530e | powerflex                      | True      |
+
+```
+
```

Comment:
do a show of the volume type to show the mapping of the volume_backend_name with the one provided in customServiceConfig section of the CR.